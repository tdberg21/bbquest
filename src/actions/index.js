export const addRestaurants = (restaurants) => ({
  type: 'ADD_RESTAURANTS',
  restaurants
});

export const loginUser = (username, id) => ({
  type: 'LOGIN_USER',
  username,
  id
});

