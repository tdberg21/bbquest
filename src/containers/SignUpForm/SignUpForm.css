
.signup-form-header {
  margin: 0;
}