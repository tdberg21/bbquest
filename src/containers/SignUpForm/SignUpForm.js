import React, { Component } from 'react';
import connect from 'react-redux';
import { Link } from 'react-router-dom';
import './SignUpForm.css';


export default class SignUpForm extends Component {
  constructor () {
    super();

    this.state = {
      username: '',
      email: '',
      password: ''
    };
  }

  handleChange = (event) => {
    const { name, value } = event.target;
    this.setState({
      [name]: value
    });
  }

  handleSubmit = (event) => {
    event.preventDefault();
    console.log('submit');
  }

  render () {
    return (
      <div className='signup-form-container'>
        <form
          className='login-form'
          onSubmit={this.handleSubmit}
        >
          <h3 className='signup-form-header'>Please signup to search and track barbecue!</h3>
          <div className='inputs'>
            <input
              className='username-field'
              aria-label='Please Enter Your Username'
              placeholder='username'
              type='text'
              name='username'
              value={this.state.username}
              onChange={this.handleChange}
            />
            <input
              className='email-field'
              aria-label='Please Enter Your Email'
              placeholder='email'
              type='text'
              name='email'
              value={this.state.email}
              onChange={this.handleChange}
            />
            <input
              className='password-field'
              aria-label='Please Enter Your Password'
              placeholder='password'
              type='password'
              name='password'
              value={this.state.password}
              onChange={this.handleChange}
            />
            <button
              className='signup-button'
              aria-label='Sign up your account'
            >
              Signup
            </button>
          </div>
          <Link to='/login'>
            Already have an account?
          </Link>
        </form>
      </div>
    );
  }
}