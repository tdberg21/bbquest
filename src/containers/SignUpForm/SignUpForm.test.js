import React from 'react';
import { shallow } from 'enzyme';
import { SignUpForm, mapDispatchToProps } from './SignUpForm';
import { loginUser } from '../../actions';

describe('SIGN UP FORM TESTS', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<SignUpForm />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should update state when handleChange is invoked', () => {
    const wrapper = shallow(<SignUpForm />);
    const mockEvent = { target: { name: 'username', value: 'Tory' } };

    wrapper.instance().handleChange(mockEvent);

    expect(wrapper.state('username')).toEqual('Tory');

  });


  describe('mapDispatchToProps', () => {
    it('calls dispatch with loginUser action', async () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = loginUser('Tory', 'td@g.com', 'password');
      const mappedProps = mapDispatchToProps(mockDispatch);
      
      mappedProps.loginUser('Tory', 'td@g.com', 'password');

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });
});