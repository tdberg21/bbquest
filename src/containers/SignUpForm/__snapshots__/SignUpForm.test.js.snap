// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SIGN UP FORM TESTS should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignUpForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        className="login-form"
        onSubmit={[Function]}
>
        <div
                className="inputs"
        >
                <label
                        for="username"
                >
                        Username:
                </label>
                <input
                        aria-label="Please Enter Your Username"
                        className="username-field input-fields"
                        id="username"
                        name="username"
                        onChange={[Function]}
                        type="text"
                        value=""
                />
                <label
                        for="email"
                >
                        Email:
                </label>
                <input
                        aria-label="Please Enter Your Email"
                        className="email-field input-fields"
                        id="email"
                        name="email"
                        onChange={[Function]}
                        type="text"
                        value=""
                />
                <label
                        for="password"
                >
                        Password:
                </label>
                <input
                        aria-label="Please Enter Your Password"
                        className="password-field input-fields"
                        id="password"
                        name="password"
                        onChange={[Function]}
                        type="password"
                        value=""
                />
                <button
                        aria-label="Sign up your account"
                        className="login-button"
                >
                        Signup
                </button>
        </div>
        <Link
                replace={false}
                to="/login"
        >
                Already have an account?
        </Link>
</form>,
      "className": "login-form-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="inputs"
>
            <label
                        for="username"
            >
                        Username:
            </label>
            <input
                        aria-label="Please Enter Your Username"
                        className="username-field input-fields"
                        id="username"
                        name="username"
                        onChange={[Function]}
                        type="text"
                        value=""
            />
            <label
                        for="email"
            >
                        Email:
            </label>
            <input
                        aria-label="Please Enter Your Email"
                        className="email-field input-fields"
                        id="email"
                        name="email"
                        onChange={[Function]}
                        type="text"
                        value=""
            />
            <label
                        for="password"
            >
                        Password:
            </label>
            <input
                        aria-label="Please Enter Your Password"
                        className="password-field input-fields"
                        id="password"
                        name="password"
                        onChange={[Function]}
                        type="password"
                        value=""
            />
            <button
                        aria-label="Sign up your account"
                        className="login-button"
            >
                        Signup
            </button>
</div>,
          <Link
            replace={false}
            to="/login"
>
            Already have an account?
</Link>,
        ],
        "className": "login-form",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                for="username"
>
                Username:
</label>,
              <input
                aria-label="Please Enter Your Username"
                className="username-field input-fields"
                id="username"
                name="username"
                onChange={[Function]}
                type="text"
                value=""
/>,
              <label
                for="email"
>
                Email:
</label>,
              <input
                aria-label="Please Enter Your Email"
                className="email-field input-fields"
                id="email"
                name="email"
                onChange={[Function]}
                type="text"
                value=""
/>,
              <label
                for="password"
>
                Password:
</label>,
              <input
                aria-label="Please Enter Your Password"
                className="password-field input-fields"
                id="password"
                name="password"
                onChange={[Function]}
                type="password"
                value=""
/>,
              <button
                aria-label="Sign up your account"
                className="login-button"
>
                Signup
</button>,
            ],
            "className": "inputs",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Username:",
                "for": "username",
              },
              "ref": null,
              "rendered": "Username:",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "aria-label": "Please Enter Your Username",
                "className": "username-field input-fields",
                "id": "username",
                "name": "username",
                "onChange": [Function],
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Email:",
                "for": "email",
              },
              "ref": null,
              "rendered": "Email:",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "aria-label": "Please Enter Your Email",
                "className": "email-field input-fields",
                "id": "email",
                "name": "email",
                "onChange": [Function],
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Password:",
                "for": "password",
              },
              "ref": null,
              "rendered": "Password:",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "aria-label": "Please Enter Your Password",
                "className": "password-field input-fields",
                "id": "password",
                "name": "password",
                "onChange": [Function],
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "aria-label": "Sign up your account",
                "children": "Signup",
                "className": "login-button",
              },
              "ref": null,
              "rendered": "Signup",
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Already have an account?",
            "replace": false,
            "to": "/login",
          },
          "ref": null,
          "rendered": "Already have an account?",
          "type": [Function],
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          className="login-form"
          onSubmit={[Function]}
>
          <div
                    className="inputs"
          >
                    <label
                              for="username"
                    >
                              Username:
                    </label>
                    <input
                              aria-label="Please Enter Your Username"
                              className="username-field input-fields"
                              id="username"
                              name="username"
                              onChange={[Function]}
                              type="text"
                              value=""
                    />
                    <label
                              for="email"
                    >
                              Email:
                    </label>
                    <input
                              aria-label="Please Enter Your Email"
                              className="email-field input-fields"
                              id="email"
                              name="email"
                              onChange={[Function]}
                              type="text"
                              value=""
                    />
                    <label
                              for="password"
                    >
                              Password:
                    </label>
                    <input
                              aria-label="Please Enter Your Password"
                              className="password-field input-fields"
                              id="password"
                              name="password"
                              onChange={[Function]}
                              type="password"
                              value=""
                    />
                    <button
                              aria-label="Sign up your account"
                              className="login-button"
                    >
                              Signup
                    </button>
          </div>
          <Link
                    replace={false}
                    to="/login"
          >
                    Already have an account?
          </Link>
</form>,
        "className": "login-form-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="inputs"
>
              <label
                            for="username"
              >
                            Username:
              </label>
              <input
                            aria-label="Please Enter Your Username"
                            className="username-field input-fields"
                            id="username"
                            name="username"
                            onChange={[Function]}
                            type="text"
                            value=""
              />
              <label
                            for="email"
              >
                            Email:
              </label>
              <input
                            aria-label="Please Enter Your Email"
                            className="email-field input-fields"
                            id="email"
                            name="email"
                            onChange={[Function]}
                            type="text"
                            value=""
              />
              <label
                            for="password"
              >
                            Password:
              </label>
              <input
                            aria-label="Please Enter Your Password"
                            className="password-field input-fields"
                            id="password"
                            name="password"
                            onChange={[Function]}
                            type="password"
                            value=""
              />
              <button
                            aria-label="Sign up your account"
                            className="login-button"
              >
                            Signup
              </button>
</div>,
            <Link
              replace={false}
              to="/login"
>
              Already have an account?
</Link>,
          ],
          "className": "login-form",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <label
                  for="username"
>
                  Username:
</label>,
                <input
                  aria-label="Please Enter Your Username"
                  className="username-field input-fields"
                  id="username"
                  name="username"
                  onChange={[Function]}
                  type="text"
                  value=""
/>,
                <label
                  for="email"
>
                  Email:
</label>,
                <input
                  aria-label="Please Enter Your Email"
                  className="email-field input-fields"
                  id="email"
                  name="email"
                  onChange={[Function]}
                  type="text"
                  value=""
/>,
                <label
                  for="password"
>
                  Password:
</label>,
                <input
                  aria-label="Please Enter Your Password"
                  className="password-field input-fields"
                  id="password"
                  name="password"
                  onChange={[Function]}
                  type="password"
                  value=""
/>,
                <button
                  aria-label="Sign up your account"
                  className="login-button"
>
                  Signup
</button>,
              ],
              "className": "inputs",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Username:",
                  "for": "username",
                },
                "ref": null,
                "rendered": "Username:",
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "aria-label": "Please Enter Your Username",
                  "className": "username-field input-fields",
                  "id": "username",
                  "name": "username",
                  "onChange": [Function],
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Email:",
                  "for": "email",
                },
                "ref": null,
                "rendered": "Email:",
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "aria-label": "Please Enter Your Email",
                  "className": "email-field input-fields",
                  "id": "email",
                  "name": "email",
                  "onChange": [Function],
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Password:",
                  "for": "password",
                },
                "ref": null,
                "rendered": "Password:",
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "aria-label": "Please Enter Your Password",
                  "className": "password-field input-fields",
                  "id": "password",
                  "name": "password",
                  "onChange": [Function],
                  "type": "password",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "aria-label": "Sign up your account",
                  "children": "Signup",
                  "className": "login-button",
                },
                "ref": null,
                "rendered": "Signup",
                "type": "button",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Already have an account?",
              "replace": false,
              "to": "/login",
            },
            "ref": null,
            "rendered": "Already have an account?",
            "type": [Function],
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
